### ISSUE TITLE
Incorrect code location for functions defined in standard input

### ISSUE THREAD

**Issue by d-torrance** [Macaulay2/M2#3415]
Since #3403:

```m2
i1 : f = x -> x^2

o1 = f

o1 : FunctionClosure

i2 : code f

o2 = stdio:1:6-1:11: --source code:
     -- This is the beginning of your Macaulay2 log stored at /home/profzoom/.Macaulay2/history.m2
```

I suppose the solution would be to keep track of how long the history is when we start M2 and add that to the line number when finding the code.

**Comment by mahrud** [comment]
Ah yes I should have noticed this from the getHistory lines in code.m2.

**Comment by mahrud** [comment]
This is actually more complicated than I expected.
I added a call to `where_history` in this [commit](https://github.com/mahrud/M2/commit/9eed58c5d9c5ad317c672f6223cba8243fd1a42e), but:
1. it seems to return zero if it's called before the first prompt, even if it's after `readHistory` apparently!
2. it seems to be off by an inconsistent number (up to 4 even) from the right number to get `code` working in your example.

I'm not sure if I have time to debug this. We can disable history if this is too complicated to resolve.

**Comment by d-torrance** [comment]
I think I figured it out -- see #3416.

**Comment by d-torrance** [comment]
Closed by #3416
